lib/features/session/domain/usecases/get_session_interactor.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:core/core.dart';
+import 'package:dartz/dartz.dart';
+import 'package:tmail_ui_user/features/session/domain/repository/session_repository.dart';
+import 'package:tmail_ui_user/features/session/domain/state/get_session_state.dart';
+
+class GetSessionInteractor {
+  final SessionRepository sessionRepository;
+
+  GetSessionInteractor(this.sessionRepository);
+
+  Future<Either<Failure, Success>> execute() async {
```

Comment:
Why not return a stream.
Loading -> State ...

Reply:
The processing of this task occurs on a loading screen. So I don't think the state needs to be concerned here.